File: BOJ/2_WEEKS/013_14252.cpp
#include<iostream>
#include<algorithm> //sort
using namespace std;

int n, cnt, a[51];

int getGCD(int a, int b) {
	if (a == 0) {
        return b;
    }
    
	return getGCD(b % a, a);
}

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);

	cin >> n;

	for (int i = 0; i < n; i++) {
		cin >> a[i];
	}

	sort(a, a + n);

	for (int i = 0; i < n - 1; i++) {
		if (getGCD(a[i], a[i + 1]) == 1) continue;

		for (int j = a[i] + 1; j < a[i + 1]; j++) {
			if (getGCD(a[i], j) == 1) {
				if (getGCD(j, a[i + 1]) == 1) {
					cnt++; // 1개의 수로 공약수열 완성
					break;
				}
			}

			//j가 끝에 도달했지만 공약수열이 완성되지 않는 경우
			if (j == a[i + 1] - 1) {
				cnt += 2; //https://en.wikipedia.org/wiki/B%C3%A9zout%27s_identity
				break;
			}
		}
	}

	cout << cnt;

	return 0;
}

File: baekjoon/sensei/java/013_14252.java
/**
 * 시간복잡도 O(N * 10^5 * log(10^5))
 */

import java.util.*;
import java.io.*;

public class Main {
    static CustomScanner cs = new CustomScanner();
    static int n, ret;

    public static void main(String[] args) throws IOException {
        n = cs.nextInt();

        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = cs.nextInt();
        }

        Arrays.sort(arr);

        for (int i = 0; i < n - 1; i++) {
            if (getGcd(arr[i], arr[i + 1]) == 1) {
                continue;
            }

            for (int j = arr[i] + 1; j < arr[i + 1]; j++) {
                if (getGcd(arr[i], j) == 1) {
                    if (getGcd(j, arr[i + 1]) == 1) {
                        ret++;
                        break;
                    }
                }

                if (j == arr[i + 1] - 1) {
                    ret += 2; // https://en.wikipedia.org/wiki/B%C3%A9zout%27s_identity
                }
            }
        }

        System.out.println(ret);
    }

    static int getGcd(int a, int b) {
        if (a == 0) {
            return b;
        }

        return getGcd(b % a, a);
    }

    static class CustomScanner {
        private final BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        private StringTokenizer st;

        String next() throws IOException {
            while (st == null || !st.hasMoreTokens()) {
                String line = br.readLine();

                if (line.isEmpty()) {
                    continue;
                }

                st = new StringTokenizer(line);
            }

            return st.nextToken();
        }

        Integer nextInt() throws IOException {
            String s = next();

            return Integer.parseInt(s);
        }

        Long nextLong() throws IOException {
            String s = next();

            return Long.parseLong(s);
        }
    }
}
